en/02_Developer_Guides/08_Performance/06_ORM.md

```diff
@@ -49,3 +49,15 @@ SilverStripe\ORM\Connect\DBSchemaManager:
 ```
 
 You can always manually trigger a check and repair (e.g. in a [`BuildTask`](api:SilverStripe/Dev/BuildTask)) by calling [`DB::check_and_repair_table()`](api:SilverStripe\ORM\DB::check_and_repair_table()). This ignores the above configuration.
+
+## Changing `ClassName` column from enum to varchar {#classname-varchar}
+
+On websites with very large database tables it can take a long time to run `dev/build`, which can be a problem when deploying changes to production. This is because the `ClassName` column is an `Enum` type which requires an a `ALTER TABLE` query to be run affecting every row whenever there is a new valid value for the column.
+
+You may wish to change the `ClassName` column to a `Varchar` type which remove the need to run `ALTER TABLE` whenever there is a new valid value. Enabling this will result in a trade-off where the size of the database will increase by approximately 6mb per 100,000 rows. There will also be a very slow initial `dev/build` as all of the `ClassName` columns are switched to `Varchar`. To enable this, add the following configuration:
```

Comment:
Just like with the changelog - please add a rough estimate of the expected performance increase so people can make an informed decision about this.

Reply:
I don't think I can provide any meaningful measurements here.

The performance difference between enum vs varchar is technically unlimited if you look at that one column in isolation, because with varchar you simply don't run ALTER TABLE queries.

In terms of the % of dev/build time spent doing ALTER TABLE queries, is be totally different of different websites.  This means I there's no way I can say "dev/build time is x% faster using varchars".

I created a [module](https://github.com/emteknetnz/silverstripe-dev-build-benchmark) that shows the % of dev/build query time spent on ALTER TABLE queries, and on a couple of different sites (a barebones local, and a deployed test setup with some data) I got completely different results. Real world sites will have a completely different % again.

Reply:
This doesn't give me much confidence that it does improve performance, if you're saying it's not reliably measurable.

Reply:
Well it's removing queries during dev/build, however the % that those queries take up has massive variance between websites.  On some websites it will be negliable, in others it could well be 10s of minutes.

I'm not sure how I'm supposed to provide a measurement here

This solution is very much only for a niche of websites with huge database tables that struggle with deployments. As I've mentioned above I've provided a module that can help qualify if this solution may be useful as it will show roughly the potential time saving from removing the ALTER TABLE queries.

Reply:
You were able to provide an estimate of the amount of extra db space required, which will also vary from project to project. You did this by equating it to an amount of space per x records.
Surely a similar thing can be done with performance? For each X records, roughly Y seconds will be saved? It doesn't have to be exact, just a ballpark based on observations in your local.

Reply:
Extra db space was very easy measure as what I provided is simply the extra bytes for an averaged sized ClassName which I defined myself. There is no variance there.

However I cannot provide a estimate for the total bytes saved for a project, because there will be massive variance between projects.

Reply:
Will say an ALTER TABLE query with 10 million took 28.52 seconds on my local